captum/optim/_core/objectives.py

```diff
@@ -143,10 +142,8 @@ def optimize(
                 (-1 * loss_value.mean()).backward()
                 optimizer.step()
                 step += 1
-        except (Exception, BaseException) as e:
+        finally:
             self.cleanup()
-            raise e
-        self.cleanup()
         return history

```

Comment:
Suggestion: Do we want to return `history` as a PyTorch tensor instead of a NumPy array?

Reply:
I'd prefer returning PT tensor as well. This will be consistent with Captum library.

Reply:
Updated! Now returning Tensor